R-package/tests/testthat/test_learning_to_rank.R

```diff
@@ -0,0 +1,94 @@
+context("Learning to rank")
```

Comment:
totally agree. Since this is currently a user-facing bug (i.e. the functionality is already incorrect on `master`, #2572 ), I wanted to get something that I think is a fix up as fast as possible. I'm happy to keep expanding these tests and pushing on this PR until @Laurae2 has a chance to review the changes to `lgb.cv()`, but once we get that approval I don't think we should delay merging this for the sake of more testing.

Reply:
Sure! I think adding a check for adequate values of NDCG will be enough for now.

Reply:
Thanks for the  feedback @StrikerRUS , very good idea. I've added more to these tests in https://github.com/microsoft/LightGBM/pull/2573/commits/f14458bd3678d8b8883f8a8c38a750c769b8f223

I'm testing the  following things for  learning-to-rank with  `lambdarank`.  I think this should be sufficient.

`lgb.train()`

* returns an `lgb.Booster`  object, with the following properties
    - `max_feature_idx` matching dimensions of the training data
    -  one set  of `eval_results` per metric (3 in these tests, since I had  `ndcg_at = "1,2,3"`
     - all evaluation results have reasonable values (between 0.0 and 1.0 since they're nDCG)

`lgb.cv()`

* returns an `lgb.CVBooster` object, with the following properties
    - `boosters`, containing a list of length `nfold`  of  `lgb.Booster`
    - `best_iter`  with a reasonable value  (>1, <=  `nrounds`)
    - `best_score`  is based on the first metric (`ncdg@1`, in this case)
    -  one set  of `eval_results` per metric (3 in these tests, since I had  `ndcg_at = "1,2,3"`. For which:
         - `eval` and `eval_err`  are arrays with length `nrounds`  (since  I did not use  early stopping)
          - all `eval`  values  have reasonable values (between 0.0 and 1.0 since they're nDCG)


@guolinke or  @Laurae2 could I get one more review of the changes in this PR from one of you?

Reply:
@jameslamb 

> all evaluation results have reasonable values (between 0.0 and 1.0 since they're nDCG)

I think that tests can be more strict to the final metric value in that aspect to ensure that training actually happens. Refer to Python tests as an example:
https://github.com/microsoft/LightGBM/blob/301402c83bc732f4ec41135cc5e5f6f5984348d8/tests/python_package_test/test_sklearn.py#L117-L118

Reply:
oh sure I can go down to that level. I avoided it because I thought it would make the tests too fragile (even with setting the seed), but if it's been working on the Python side then I'm ok with it. Let me try that.

Reply:
thanks @StrikerRUS , I've made these tests much more specific in https://github.com/microsoft/LightGBM/pull/2573/commits/ead17bcc6ddb06fa506071e76ce0bba213101f8c. Could you take a look?

Reply:
fixed in  https://github.com/microsoft/LightGBM/pull/2573/commits/ee037c9a2f32415c7bd7e63a4d1c093687d110e8 (had to rebase to `master` and force push with other R PRs that were just merged)

Reply:
@jameslamb Thanks you! Tests are more reliable now!